### DISCUSSION TITLE
General Ideas and features

### DISCUSSION THREAD

**Discussion by gordlea** [gordlea/fireboard2mqtt#39] [Ideas]
I think in version 3.1 I will try to add the following:

### Support for alarms as sensors
If a channel has an alarm configured, we can have 2 binary_sensors:

* Channel X Alert Too Cold: [binary_sensor with device_class](https://www.home-assistant.io/integrations/binary_sensor/#device-class) Cold
* Channel X Alert Too Hot: [binary_sensor with device_class](https://www.home-assistant.io/integrations/binary_sensor/#device-class) Hot

Anyone else have any idea's for features/sensors that could be added, or use cases that aren't quite covered by the current functionality?

**Comment by alexownsall** [comment]
I will think of potential functionality. Do you have a link to the API spec? Might help seeing some of the data that is available to spring ideas. Also, I agree about keeping the debug option on as default for now to troubleshoot.

**Comment by mattdevo1** [comment]
If supported by the API, it would be great if we could change the drive set point and control channel from HA.

Honestly, I could stop using the FireBoard app at that point I think.

Keep up the great work man, thank you for building this add-on. Awesome stuff!

**Comment by gordlea** [comment]
Here are the api docs: https://docs.fireboard.io/app/api.html

Unfortunately it's read only. 

My biggest wish is a local api where you could query the device for data directly, without rate limits.

It appears to actually use mqtt internally which you can see if you go to (https://fireboard.io/sessions/#/) and use the browser dev tools to inspect the websocket network requests (it has the header: `Sec-WebSocket-Protocol: mqttv3.1`.

**Comment by scottshanafelt** [comment]
I was also going to mention adjusting the setpoint, but understand it's read only.  But other than that it seems great!  Love the alarms feature idea.

My current use cases for getting this into Home Assistant:
- If pit temp drops too low below setpoint for too long (out of pellets, flame out, etc) blow my phone up
- Send tts announcements to my smart speakers when sensor temps reach desired temp.

That second one would be a lot more convenient if it worked off of the fireboard alert system (rather than adjusting automations in HA).

Thank you so much for this project.  Not sure how many people are nerdy enough to use it, but it's huge for those of us who are!

**Reply by gordlea** [reply]
No problem! It's been fun to build it. And these use cases are really helpful 👍
